docs/pages/machine-workload-identity/machine-id/deployment/kubernetes-oidc.mdx

```diff
@@ -0,0 +1,454 @@
+cluster that supports OIDC. The Kubernetes OIDC join method is a good fit for
+use inside Kubernetes clusters on a cloud platform that can be configured to
+provide a public OpenID Connect (OIDC) endpoint, including EKS, AKS, and GKE.
+
+## How it works
+
+In the setup we demonstrate in this guide, `tbot` runs as a Kubernetes
+deployment. It writes output credentials to a Kubernetes secret, which can then
+be mounted in the pods that need to use the credentials. While `tbot` can also
+run as a sidecar within the same pod as the service that needs to use the
+credentials it generates, we recommend running `tbot` as a standalone deployment
+due the limited support Kubernetes has for sidecars.
+
+In this guide, we demonstrate the `kubernetes` join method using its OIDC
+support, in which `tbot` proves its identity to the Teleport Auth Service by
+presenting a JSON web token (JWT) signed by the platform OIDC issuer. This JWT
+is projected into pods by Kubernetes, and identifies the service account, the
+pod, and the namespace in which `tbot` is running. The Teleport Auth Service
+checks the signature of the JWT against the Kubernetes OIDC provider's published
+signing keys to verify the pod identity.
+
+<Admonition type="warning" title="OIDC Support">
+Not all Kubernetes providers support [Service Account Issuer Discovery][said],
+which is required for this guide. If your provider does not support this
+feature, refer to our [Kubernetes with Static JWKS guide](./kubernetes.mdx)
+instead.
+</Admonition>
+
+<details>
+<summary>Using another join method</summary>
+
+When deploying `tbot` to a Teleport cluster, it is generally recommended to use
+the `kubernetes` join method. This will work with most Kubernetes clusters.
+The guide that follows will demonstrate configuring this join method.
+
+However, when using certain cloud Kubernetes services, it is possible to use the
+join method associated with that platform rather than the `kubernetes` join
+method. This may be beneficial if you wish to manage the joining of `tbot`
+within the Kubernetes clusters and on standard VMs on the same platform with
+a single join token. These services are:
+
+- Google Kubernetes Engine: Where
+  [GCP Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)
+  is configured for the cluster, it is possible to use the `gcp` join method.
+  See the [GCP Platform Guide](./gcp.mdx) for further information.
+- Amazon Elastic Kubernetes Service: Where
+  [IAM Roles for Service Accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)
+  is configured for the cluster, it is possible to use the `iam` join method.
+  See the [AWS Platform Guide](./aws.mdx) for further information.
+
+It is recommended to use Kubernetes OIDC joining on Azure (AKS) whenever
+possible.
+
+</details>
+
+## Prerequisites
+
+- A running Teleport cluster, version 17.7.3 or above.
```

Comment:
I would also list the v18 version here. Don't want people who land on this page to assume that 18.0.0 is newer than 17.7.3 and then get confused when it doesn't work.

Reply:
That's a fair point. I've taken a stab at including both variants - it's surprisingly tricky to phrase multiple version requirements concisely, but I think it reads okay.